src/runtime_config/actions.js

```diff
@@ -148,23 +172,46 @@ const actions = new Map([
       },
 
       function (args) {
-        let user_id = ccf.pemToId(args.cert);
-        let raw_user_id = ccf.strToBuf(user_id);
+        let userId = ccf.pemToId(args.cert);
+        let rawUserId = ccf.strToBuf(userId);
 
-        if (ccf.kv["ccf.gov.users.certs"].has(raw_user_id)) {
-          return; // Idempotent
-        }
+        ccf.kv["public:ccf.gov.users.certs"].set(
+          rawUserId,
+          ccf.strToBuf(args.cert)
+        );
 
-        ccf.kv["ccf.gov.users.certs"].set(raw_user_id, ccf.strToBuf(args.cert));
+        if (args.user_data !== null && args.user_data !== undefined) {
+          ccf.kv["public:ccf.gov.users.info"].set(
+            rawUserId,
+            ccf.jsonCompatibleToBuf(args.user_data)
+          );
+        } else {
+          ccf.kv["public:ccf.gov.users.info"].delete(rawUserId);
+        }
+      }
+    ),
+  ],
+  [
+    "set_user_data",
+    new Action(
+      function (args) {
+        return (
+          typeof args.user_id === "string" && typeof args.user_data === "object"
+        );
+      },
+      function (args) {
+        const userId = ccf.strToBuf(args.user_id);
 
-        if (args.user_data != null) {
-          if (ccf.kv["ccf.gov.users.info"].has(raw_user_id)) {
-            throw new Error(`User info for ${user_id} already exists`);
-            // Internal error
-          }
+        const rawUserInfo = ccf.kv["public:ccf.gov.users.info"].get(userId);
+        if (rawUserInfo === undefined) {
+          return; // Idempotent if proposal deletes user data
```

Comment:
That doesn't seem correct? If there is no user_data, but the proposal wants to set it, we should set it?

I also don't understand the point of set_user_data now that set_user is idempotent.